client/txn.go

```diff
@@ -289,6 +279,25 @@ func (txn *Txn) Run(b *Batch) error {
 	return sendAndFill(txn.send, b)
 }
 
+func (txn *Txn) commit() error {
+	return txn.sendEndTxnCall(true /* commit */)
+}
+
+func (txn *Txn) cleanup(err error) {
+	if err != nil {
+		if replyErr := txn.Rollback(); replyErr != nil {
+			log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+		}
+	}
+}
+
+// CommitNoCleanup is the same as Commit but will not attempt to clean
+// up on failure. It is exposed only for use in txn_correctness_test.go
+// because those tests manipulate transaction state at a low level.
+func (txn *Txn) CommitNoCleanup() error {
```

Comment:
Code cleanups are undeniably useful, but they come with associated risk. 

Have you fixed a bug with this change? You can answer this question by writing a new test that fails before the change and passes afterwards. This change contains no such test so I'm assuming no bugs were fixed. 

Have you introduced a bug? This is more difficult to answer. I assume the current tests still pass, but we all know that test coverage is never complete. Additional tests would be reassuring, but perhaps the existing coverage already appears sufficient. I'm going to have to examine this change more carefully when my mind is fresher.